### PR TITLE
Fixed DatabaseFeatures.supports_index_column_ordering and related tests with MyISAM storage engine.

### PR DESCRIPTION
**Pull request by felixxm**

### CODE REVIEW

File: tests/indexes/tests.py [hunk 1]
```diff
@@ -336,7 +336,7 @@ def test_no_index_for_foreignkey(self):
                 ArticleTranslation._meta.db_table,
             )
         if storage != "InnoDB":
-            self.skip("This test only applies to the InnoDB storage engine")
+            self.skipTest("This test only applies to the InnoDB storage engine")
```
Comment (felixxm) [inline tests/indexes/tests.py:339]: It's a typo, `TransactionTestCase` doesn't have `skip()` method.

File: django/db/backends/mysql/features.py [hunk 2]
```diff
@@ -328,6 +328,8 @@ def can_introspect_json_field(self):
 
     @cached_property
     def supports_index_column_ordering(self):
+        if self._mysql_storage_engine != "InnoDB":
+            return False
```
Comment (felixxm) [inline django/db/backends/mysql/features.py:332]: MySQL and MariaDB support column order in indexes only with `InnoDB` storage engine.

File: django/db/models/indexes.py [hunk 3]
```diff
@@ -111,6 +111,10 @@ def create_sql(self, model, schema_editor, using="", **kwargs):
                 for field_name, _ in self.fields_orders
             ]
             col_suffixes = [order[1] for order in self.fields_orders]
+            if schema_editor.connection.features.supports_index_column_ordering:
+                col_suffixes = [order[1] for order in self.fields_orders]
+            else:
+                col_suffixes = [''] * len(self.fields_orders)
```
Comment (felixxm) [inline django/db/models/indexes.py:117]: Most databases ignore column ordering when it's not supported, unfortunately MySQL/MariaDB with `MyISAM` storage engine raise an exception:
```
django.db.utils.OperationalError: (1178, "The storage engine for the table doesn't support descending indexes")
```

**Review by carltongibson — APPROVED** [review]
OK, super. Thanks

**Comment by felixxm** [comment]
Thanks :+1:

**Pull request merged**